robosuite/utils/mjmod.py

```diff
@@ -1397,85 +1398,521 @@ def bitmap(self):
+        ):
+            for name in group_randomize_names:
+                # Randomize all parameters associated with this element
+                for attr, default_val in group_defaults[name].items():
+                    val = copy.copy(default_val)
+                    settings = group_randomizations[attr]
+                    if settings["randomize"]:
+                        # Randomize accordingly, and clip the final perturbed value
+                        perturbation = np.random.rand() if type(val) in {int, float} else np.random.rand(*val.shape)
+                        perturbation = settings["perturbation"] * (-1 + 2 * perturbation)
+                        val = val + perturbation if settings["type"] == "size" else val * (1.0 + perturbation)
+                        val = np.clip(val, *settings["clip"])
+                    # Modify this value
+                    self.mod(name=name, attr=attr, val=val)
+
+        # Make sure changes propagate in sim
+        self.update()
 
-        ret = opt_attr if opt_attr is not None else model_attr
-        if callable(ret):
-            def r(*args):
-                return ret(*args)
-            return r
+    def update_sim(self, sim):
+        """
+        In addition to super method, reset all internal values
+
+        Args:
+            sim (MjSim): MjSim object
+        """
+        super().update_sim(sim=sim)
+        self.save_defaults()
 
-        return ret
+    def update(self):
+        """
+        Propagates the changes made up to this point through the simulation
+        """
+        self.sim.forward()
 
-    def mod(self, **kwargs):
+    def mod(self, name, attr, val):
         """
-        Method to actually mod. Assumes passing in keyword arguments with key being the parameter to
-        modify and the value being the value to set
-        Feel free to add more as we see fit.
+        General method to modify dynamics parameter @attr to be new value @val, associated with element @name.
 
         Args:
-            **kwargs (dict): Physical parameters to actually modify mid-sim
+            name (str): Name of element to modify parameter. This can be a body, geom, or joint name. If modifying
+                an opt parameter, this should be set to None
+            attr (str): Name of the dynamics parameter to modify. Valid options are self.dynamics_parameters
+            val (int or float or n-array): New value(s) to set for the given dynamics parameter. The type of this
+                argument should match the expected type for the given parameter.
         """
-        for to_mod in kwargs:
-            val = kwargs[to_mod]
+        # Make sure specified parameter is valid, and then modify it
+        assert attr in self.dynamics_parameters, "Invalid dynamics parameter specified! Supported parameters are: {};" \
+                                                 " requested: {}".format(self.dynamics_parameters, attr)
+        getattr(self, f"mod_{attr}")(name, val)
```

Comment:
cool trick - consider adding a comment here so it's more clear that this is a way to call one of the below methods by name, based on the variable

Reply:
Done!